p2p/protocols/accounting.go

```diff
@@ -108,87 +110,87 @@ func NewAccounting(balance Balance, po Prices) *Accounting {
-	//instantiate the metrics
-	mBalanceCredit = metrics.NewRegisteredCounterForced("account.balance.credit", registry)
-	mBalanceDebit = metrics.NewRegisteredCounterForced("account.balance.debit", registry)
-	mBytesCredit = metrics.NewRegisteredCounterForced("account.bytes.credit", registry)
-	mBytesDebit = metrics.NewRegisteredCounterForced("account.bytes.debit", registry)
-	mMsgCredit = metrics.NewRegisteredCounterForced("account.msg.credit", registry)
-	mMsgDebit = metrics.NewRegisteredCounterForced("account.msg.debit", registry)
-	mPeerDrops = metrics.NewRegisteredCounterForced("account.peerdrops", registry)
-	mSelfDrops = metrics.NewRegisteredCounterForced("account.selfdrops", registry)
-	//create the DB and start persisting
-	return NewAccountingMetrics(registry, reportInterval, path)
+	// create an empty registry
+	MetricsRegistry = metrics.NewRegistry()
+	// instantiate the metrics
+	mBalanceCredit = metrics.NewRegisteredCounterForced("account.balance.credit", MetricsRegistry)
+	mBalanceDebit = metrics.NewRegisteredCounterForced("account.balance.debit", MetricsRegistry)
+	mBytesCredit = metrics.NewRegisteredCounterForced("account.bytes.credit", MetricsRegistry)
+	mBytesDebit = metrics.NewRegisteredCounterForced("account.bytes.debit", MetricsRegistry)
+	mMsgCredit = metrics.NewRegisteredCounterForced("account.msg.credit", MetricsRegistry)
+	mMsgDebit = metrics.NewRegisteredCounterForced("account.msg.debit", MetricsRegistry)
+	mPeerDrops = metrics.NewRegisteredCounterForced("account.peerdrops", MetricsRegistry)
+	mSelfDrops = metrics.NewRegisteredCounterForced("account.selfdrops", MetricsRegistry)
+	// create the DB and start persisting
+	return NewAccountingMetrics(MetricsRegistry, reportInterval, path)
 }
 
-//Implement Hook.Send
+// Implement Hook.Send
 // Send takes a peer, a size and a msg and
-// - calculates the cost for the local node sending a msg of size to peer using the Prices interface
-// - credits/debits local node using balance interface
+//   - calculates the cost for the local node sending a msg of size to peer using the Prices interface
+//   - credits/debits local node using balance interface
 func (ah *Accounting) Send(peer *Peer, size uint32, msg interface{}) error {
-	//get the price for a message (through the protocol spec)
+	// get the price for a message (through the protocol spec)
 	price := ah.Price(msg)
-	//this message doesn't need accounting
+	// this message doesn't need accounting
 	if price == nil {
 		return nil
 	}
-	//evaluate the price for sending messages
+	// evaluate the price for sending messages
 	costToLocalNode := price.For(Sender, size)
-	//do the accounting
+	// do the accounting
 	err := ah.Add(costToLocalNode, peer)
-	//record metrics: just increase counters for user-facing metrics
+	// record metrics: just increase counters for user-facing metrics
 	ah.doMetrics(costToLocalNode, size, err)
 	return err
 }
 
-//Implement Hook.Receive
+// Implement Hook.Receive
```

Comment:
remove the comments that denote that it implements an interface method, comment them on the struct instead saying the `Accounting implements the Hook interface`. the same goes for the `Send` method.